src/cdk-constructs/IxBucket.ts

```diff
@@ -0,0 +1,45 @@
+import { Bucket } from "sst/constructs";
+import { BucketEncryption } from "aws-cdk-lib/aws-s3";
+import ixDeployConfig from "../deployConfig.js";
+
+type ConstructScope = ConstructorParameters<typeof Bucket>[0];
+type ConstructId = ConstructorParameters<typeof Bucket>[1];
+type ConstructProps = Exclude<ConstructorParameters<typeof Bucket>[2], undefined>;
+
+export class IxBucket extends Bucket {
+  constructor(
+    scope: ConstructScope,
+    id: ConstructId,
+    props: ConstructProps = {},
+  ) {
+    let bucketProps: ConstructProps = {}
+    // Merge the provided props with the defaults.
+    if (ixDeployConfig.isIxDeploy) {
+      bucketProps = {
+        ...props,
+        blockPublicACLs: true,
+        cdk: {
+          ...props.cdk,
+          bucket: {
+            ...props.cdk?.bucket,
+            enforceSSL: true,
+            encryption: BucketEncryption.S3_MANAGED,
```

Comment:
same with these, it would probably be better to include `...props.cdk?.bucket` last to allow deliberate overriding.